### ISSUE TITLE
ansible-lint tries to read files specified in lookup plugins and fails

### ISSUE THREAD

**Issue by jlgoolsbee** [ansible/ansible-lint#2459]
##### Summary

`ansible-lint` tries to read files in tasks with `ansible.builtin.ini` lookups and fails, throwing `jinja[invalid]` failures. This behavior is new; I cannot reproduce on 6.5.2 and we were/are not ignoring or skipping the previous `no-jinja-nesting` rule (I noted this on the changelog for 6.6.0 as having been replaced; it's seemingly the only relevant change in the release). I do not expect these files to exist on my local/dev machine, and `ansible-lint` has not either, until now.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
ansible --version

ansible [core 2.12.5]
  config file = /opt/atlassian/local-ansible/ansible.cfg
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/ansible212-venv/lib64/python3.8/site-packages/ansible
  ansible collection location = /root/.ansible/collections
  executable location = /usr/lib/ansible212-venv/bin/ansible
  python version = 3.8.5 (default, Feb 18 2021, 01:24:20) [GCC 7.3.1 20180712 (Red Hat 7.3.1-12)]
  jinja version = 3.1.2
  libyaml = True

ansible-lint --version

ansible-lint 6.6.0 using ansible 2.12.5

```

- ansible installation method: pip
- ansible-lint installation method: pip

##### OS / ENVIRONMENT

N/A

##### STEPS TO REPRODUCE

_debug.yml:_
```yaml
---
- name: Print a value from a file via ini lookup
  hosts: localhost
  connection: local
  tasks:
    - ansible.builtin.debug:
        msg: "SOME_VAR in some-file is: {{ lookup('ansible.builtin.ini', 'SOME_VAR', type='properties', file='/tmp/some-file') }}"
```

_/tmp/some-file:_
```
SOME_VAR=this value
```

##### Desired Behavior

`ansible-lint` should not try to read files in lookups, or at least not throw failures when those files don't exist.

##### Actual Behavior

Running `ansible-lint debug.yml` on my local machine (where the file _is not_ expected to exist):

```console
WARNING  Overriding detected file kind 'yaml' with 'playbook' for given positional argument: debug.yml
[WARNING]: Unable to find '/tmp/some-file' in expected paths (use -vvvvv to see paths)
WARNING  Listing 1 violation(s) that are fatal
jinja: An unhandled exception occurred while running the lookup plugin 'ansible.builtin.ini'. Error was a <class 'ansible.errors.AnsibleParserError'>, original message: Invalid filename: 'None'. Invalid filename: 'None' (jinja[invalid])
debug.yml:6 Task/Handler: debug msg=SOME_VAR in some-file is: {{ lookup('ansible.builtin.ini', 'SOME_VAR', type='properties', file='/tmp/some-file') }}

You can skip specific rules or tags by adding them to your configuration file:
# .config/ansible-lint.yml
warn_list:  # or 'skip_list' to silence them completely
  - jinja[invalid]  # Rule that looks inside jinja2 templates.

Finished with 1 failure(s), 0 warning(s) on 1 files.
```

Running `ansible-lint debug.yml` on a target instance (where the file _is_ expected to exist):
``` console
$ ansible-lint debug.yml
WARNING  Overriding detected file kind 'yaml' with 'playbook' for given positional argument: debug.yml
```

**Comment by ianw** [comment]
This also happens in zuul-jobs

```
roles/emit-job-header/tasks/main.yaml:2: jinja: An unhandled exception occurred while running the lookup plugin 'inventory_hostnames'. Error was a <class 'KeyError'>, original message: 'groups' (jinja[invalid])
roles/emit-job-header/tasks/main.yaml:27: jinja: An unhandled exception occurred while running the lookup plugin 'inventory_hostnames'. Error was a <class 'KeyError'>, original message: 'groups' (jinja[invalid])
```

I don't know where it got the line-2 reference from in 

https://opendev.org/zuul/zuul-jobs/src/branch/master/roles/emit-job-header/tasks/main.yaml

... afaics it doesn't reference ``inventory_hostnames```.  The line 27 reference is I guess from

```
 "{{ query('inventory_hostnames', 'all,!localhost') }}"
```

**Comment by jlgoolsbee** [comment]
Thanks for the quick fix @ssbarnea; can confirm this is resolved in 6.6.1 👍
